projects/wave-core-new/src/lib/datasets/dataset.model.ts

```diff
@@ -0,0 +1,34 @@
+import {UUID} from '../backend/backend.model';
+import {SpatialReference} from '../operators/spatial-reference.model';
+
+export interface DataSet {
+    id: InternalDataSetId; // TODO: support all Id types
+    name: string;
+    description: string;
+    result_descriptor: ResultDescriptor;
+    source_operator: string;
+}
+
+export interface InternalDataSetId {
+    Internal: UUID;
+}
+
+export interface ResultDescriptor {
+    Vector?: {
+        data_type: 'MultiPoint' | 'MultiLineString' | 'MultiPolygon';
+        spatial_reference: SpatialReference;
+        columns: { [key: string]: string}
+    };
+    Raster?: {
+        data_type: 'U8' | 'U16' | 'U32' | 'U64' | 'I8' | 'I16' | 'I32' | 'I64' | 'F32' | 'F64';
```

Comment:
Don't we have a class for these types? I guess also with deserializer.

Hmm, maybe you should distinguish between `…Dict` types which are basically our JSON representation and the type we use in WAVE.

Reply:
51f0fd1a6ee433c90556404a06b00f509760a077

Can you explain why we'dneed the distinction? In the backend we don't distinguish.

Reply:
We don't have the two types because serde does the job. Here, we have to make sure what we get upon backend requests (to type stuff from nowhere) and then convert it to something we want to work with.

Reply:
3606f66a24ee2470e7bf4d821d01717cdf9a577c